frontend/src/ui/D3Visualizations/Treemap.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useLayoutEffect, useRef } from 'react';
+import * as d3 from 'd3';
+import { TreeHierarchy } from '../../domain/hierarchy';
+
+export type TreemapProps = {
+  height: number;
+  width: number;
+  data: TreeHierarchy;
```

Comment:
Since this hierarchy structure is specific to this UI component, and you don't use TreeHierarchy.name in this viz, consider moving the type from domain into this component, and using a Transitions[] prop instead. Then you can get rid of the test data and domain files and use the transitions test data directly in your story.

Reply:
I think the data for the treemap component has to be in object form for [d3.hierarchy](https://github.com/d3/d3-hierarchy#hierarchy). Should that be done in the component file along with the type?